pkg/grpc/requestinfo/requestinfo.go

```diff
@@ -190,6 +190,14 @@ func (h *Handler) AnnotateMD(_ context.Context, req *http.Request) metadata.MD {
 	if tlsState != nil {
 		ri.VerifiedChains = ExtractCertInfoChains(tlsState.VerifiedChains)
 	}
+	return ri
+}
+
+// AnnotateMD builds a RequestInfo for a request coming in through the HTTP/1.1
+// gateway, and returns it in serialized form as GRPC metadata.
+// HTTP Request -> Metadata[B64(RequestInfo)]
+func (h *Handler) AnnotateMD(ctx context.Context, req *http.Request) metadata.MD {
```

Comment:
Let's also think about removing the context (potentially in a follow-up PR)?
```suggestion
func (h *Handler) AnnotateMD(_ context.Context, req *http.Request) metadata.MD {
```

Reply:
This function is passed to `runtime.WithMetadata()`, which requires the current function signature.